### PR TITLE
replace segment with posthog

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

File: packages/api/src/resolvers/article/index.ts [hunk 1]
```diff
@@ -491,7 +491,7 @@ export const getArticleResolver = authorized<
 //         source: 'resolver',
 //         resolver: 'setShareArticleResolver',
 //         articleId: article.id,
-//         userId: uid,
+//         distinctId: uid,
```
Comment (jacksonh) [inline packages/api/src/resolvers/article/index.ts:494]: 😬

**Review by jacksonh — APPROVED** [review]
Thanks, was there any initial analytics init code that needs to be changed, like the old identify call? I think thats all done when user requests labels now though so probably not an issue.

File: packages/api/src/resolvers/article/index.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/resolvers/article/index.ts:494]: 😆

**Comment by sywhb** [comment]
> Thanks, was there any initial analytics init code that needs to be changed, like the old identify call? I think thats all done when user requests labels now though so probably not an issue.

Yeah, it was set in the labels resolver but I think we could change it to `set_once` which will once set the user properties are not there.

```
      analytics.capture({
        distinctId: uid,
        event: 'labels',
        properties: {
          env: env.server.apiEnv,
          $set: {
            email: user.email,
            username: user.profile.username,
          },
        },
      })
```

Since Email and Username should not be changed, so I think they are safe to "set once" only.

**Pull request merged**